File: diana-web/src/main/java/org/openflexo/server/CreateChange.java
package org.openflexo.server;

import org.openflexo.fge.Drawing.DrawingTreeNode;
import org.openflexo.json.JsonVisitor;

public class CreateChange extends Change {
	private final DrawingTreeNode<?, ?> parent;
	private final DrawingTreeNode<?, ?> createdNode;

	public CreateChange(DrawingTreeNode<?, ?> createdNode, DrawingTreeNode<?, ?> parent, JsonVisitor visitor) {
		this.createdNode = createdNode;
		this.parent = parent;
	}

	public DrawingTreeNode<?, ?> getCreatedNode() {
		return createdNode;
	}

	@Override
	public void apply(ChangeManager manager) {
		System.out.println("created change applied");
	}
}

File: diana-webclient/src/app/CreateChange.ts
class CreateChange extends Change {
    private createdNode: MyNode;
    private parentId: number;
    
    public deserialize(input: any): CreateChange {
        this.createdNode = MyNode.deserialize(input.created);
        if(input.parent !== undefined) {
            this.parentId = input.parent;
        }
        return this;
    }
    
    public apply(drawing: Drawing): void {
        drawing.addNode(this.parentId, this.createdNode);
    }
}